cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2010-2020 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+#include "Box.h"
+#include "ManagedArray.h"
+#include "NeighborComputeFunctional.h"
+#include "NeighborQuery.h"
+#include "RDF.h"
+#include "StaticStructureFactorDebye.h"
+#include "utils.h"
+
+/*! \file StaticStructureFactorDebye.cc
+    \brief Routines for computing static structure factors.
+*/
+
+namespace freud { namespace diffraction {
+
+StaticStructureFactorDebye::StaticStructureFactorDebye(unsigned int bins, float k_max, float k_min)
+{
+    if (bins == 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires a nonzero number of bins.");
+    }
+    if (k_max <= 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires k_max to be positive.");
+    }
```

Comment:
Add a check for `k_min >= 0`.
```suggestion
    if (k_max <= 0)
    {
        throw std::invalid_argument("StaticStructureFactorDebye requires k_max to be positive.");
    }
    if (k_min < 0)
    {
        throw std::invalid_argument("StaticStructureFactorDebye requires k_min to be non-negative.");
    }
```